src/modules/job-manager/update.c

```diff
@@ -224,18 +379,35 @@ static void update_handle_request (flux_t *h,
     }
     /*  Process the update request. Response will be handled in update_job().
      */
-    update_job (ctx, msg, cred, job, updates);
+    update_job (update, msg, cred, job, updates);
     return;
 error:
     if (flux_respond_error (h, msg, errno, errstr) < 0)
         flux_log_error (h, "%s: flux_respond_error", __FUNCTION__);
 }
 
+static void send_error_responses (struct update *update)
+{
+    struct update_request *req = zlistx_first (update->pending_requests);
+    while (req) {
+        if (flux_respond_error (update->ctx->h,
+                                req->msg,
+                                EAGAIN,
+                                "job manager is shutting down") < 0)
+            flux_log_error (update->ctx->h,
+                            "%s: flux_respond_error",
+                            __FUNCTION__);
```

Comment:
A better log message might be "error responding to job-manager.update request".

Reply:
Sounds good. However, this is basically copying existing idiom in the job-manager, shall we clean up everywhere?

```
 grondo@pi3:~/git/flux-core.git/src/modules/job-manager$ git grep "%s: flux_respond_error"
alloc.c:        flux_log_error (h, "%s: flux_respond_error", __FUNCTION__);
alloc.c:        flux_log_error (h, "%s: flux_respond_error", __FUNCTION__);
alloc.c:        flux_log_error (h, "%s: flux_respond_error", __FUNCTION__);
annotate.c:        flux_log_error (h, "%s: flux_respond_error", __FUNCTION__);
drain.c:        flux_log_error (h, "%s: flux_respond_error", __FUNCTION__);
drain.c:        flux_log_error (h, "%s: flux_respond_error", __FUNCTION__);
drain.c:                flux_log_error (h, "%s: flux_respond_error", __FUNCTION__);
getattr.c:        flux_log_error (h, "%s: flux_respond_error", __FUNCTION__);
job-manager.c:        flux_log_error (h, "%s: flux_respond_error", __FUNCTION__);
job-manager.c:        flux_log_error (h, "%s: flux_respond_error", __FUNCTION__);
kill.c:        flux_log_error (h, "%s: flux_respond_error", __FUNCTION__);
kill.c:        flux_log_error (h, "%s: flux_respond_error", __FUNCTION__);
list.c:        flux_log_error (h, "%s: flux_respond_error", __FUNCTION__);
raise.c:        flux_log_error (h, "%s: flux_respond_error", __FUNCTION__);
raise.c:        flux_log_error (h, "%s: flux_respond_error", __FUNCTION__);
start.c:        flux_log_error (h, "%s: flux_respond_error", __FUNCTION__);
submit.c:        flux_log_error (h, "%s: flux_respond_error", __FUNCTION__);
update.c:        flux_log_error (ctx->h, "%s: flux_respond_error", __FUNCTION__);
update.c:                            "%s: flux_respond_error",
update.c:        flux_log_error (ctx->h, "%s: flux_respond_error", __FUNCTION__);
update.c:        flux_log_error (h, "%s: flux_respond_error", __FUNCTION__);
update.c:                            "%s: flux_respond_error",
urgency.c:        flux_log_error (h, "%s: flux_respond_error", __FUNCTION__);
wait.c:                flux_log_error (h, "%s: flux_respond_error", __func__);
wait.c:        flux_log_error (h, "%s: flux_respond_error", __func__);
```

Reply:
Although now I see you were pointing out this instance as a special case. Seems reasonable.

Reply:
Relying on __FUNCTION__ to provide the context for the error is not always helpful, so I've been shying away from that style of late, in new code at least.  No big deal though.

Reply:
I completely agree. It is especially maddening when you try to search for an error from the logs but get no hits because of the use of `__FUNCTION__`.

It did seem reasonable to fix that one spot, so I did so in the latest commit. The other uses of `__FUNCTION__` are pervasive (almost every return from a request handler uses this idiom in the codebase) so I didn't touch those.